chain/src/txhashset/txhashset.rs

```diff
@@ -1489,10 +1514,17 @@ pub fn zip_write(
 ) -> Result<(), Error> {
 	debug!("zip_write on path: {:?}", root_dir);
 	let txhashset_path = root_dir.clone().join(TXHASHSET_SUBDIR);
-	fs::create_dir_all(txhashset_path.clone())?;
-	zip::decompress(txhashset_data, &txhashset_path, expected_file)
-		.map_err(|ze| ErrorKind::Other(ze.to_string()))?;
-	check_and_remove_files(&txhashset_path, header)
+	fs::create_dir_all(&txhashset_path)?;
+
+	// Explicit list of files to extract from our zip archive.
+	let files = file_list(header);
+
+	// We expect to see *exactly* the paths listed above.
+	// No attempt is made to be permissive or forgiving with "alternative" paths.
+	// These are the *only* files we will attempt to extract from the zip file.
+	// If any of these are missing we will attempt to continue as some are potentially optional.
+	zip::extract_files(txhashset_data, &txhashset_path, files)?;
+	Ok(())
```

Comment:
I remember we discussed it before, still, why not to remove `?` and the last line?:)

Reply:
Just personal preference.

```
{
    one()?;
    two()?;
    three()?;
    Ok(())
}
```

reads better to me than - 

```
{
    one()?;
    two()?;
    three()
}
```

And if you need to reorder those lines or add one at the end you don't need to go reintroducing `?` (or forgetting to).

Reply:
I have been experimenting with this though recently - 

```
Ok(())
    .and_then(one())
    .and_then(two())
    .and_then(three())?
```